guide/src/project_layout.md

```diff
@@ -0,0 +1,130 @@
+# Project Layout
+
+Maturin expects a particular project layout depending on the contents of the
+package.
+
+## Pure Rust project
+
+For a pure Rust project, the structure is as expected and what you get from `cargo new`:
+
+```
+my-rust-project/
+├── Cargo.toml
+├── pyproject.toml  # required for maturing configuration
+└── src
+    ├── lib.rs  # default for library crates
+    └── main.rs  # default for binary crates
+```
+
+maturin will add a necessary `__init__.py` to the package when building the
+wheel. For convenience, this file includes the following:
+
+```python
+from .my_project import *
+
+__doc__ = .my_project.__doc__
+```
+
+such that the module functions may be called directly with:
+
+```python
+import my_project
+my_project.foo()
+```
+
+rather than:
+
+```python
+from my_project import my_project
+```
+
+N.B.: there is currently no way to tell maturin to include extra data (e.g.
+type stubs or `package_data` in setuptools) for a pure Rust project. Instead,
```

Comment:
@aganders3 This is only partially true, maturin will include type stubs for pure rust project layout if a `$module_name.pyi` file exists. For example: https://github.com/PyO3/maturin/blob/main/test-crates/pyo3-pure/pyo3_pure.pyi

Reply:
Ooh thanks for pointing this out! I played around a bit but just missed it. I will make an update to fix this note.